pgx/src/datum/numeric_support/cmp.rs

```diff
@@ -0,0 +1,126 @@
+use std::cmp::Ordering;
+
+use crate::{direct_function_call, pg_sys, AnyNumeric, Numeric};
+
+impl PartialEq<AnyNumeric> for AnyNumeric {
+    #[inline]
+    fn eq(&self, other: &AnyNumeric) -> bool {
+        unsafe {
+            direct_function_call(pg_sys::numeric_eq, vec![self.as_datum(), other.as_datum()])
+                .unwrap()
+        }
+    }
+
+    #[inline]
+    fn ne(&self, other: &AnyNumeric) -> bool {
+        unsafe {
+            direct_function_call(pg_sys::numeric_ne, vec![self.as_datum(), other.as_datum()])
+                .unwrap()
+        }
+    }
+}
+
+impl Eq for AnyNumeric {}
```

Comment:
This should document that NaN compares equal to NaN (unlike floats), which makes this implementation valid.